File: app/src/main/java/com/example/gamelister/model/db/RoomModule.java
package com.example.gamelister.model.db;

import android.content.Context;

import androidx.room.Room;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
class RoomModule {

    private static final String DB_NAME = "game-db";

    @Provides
    @Singleton
    GameDatabase provideGameDatabase(Context context) {
        return Room.databaseBuilder(
                context,
                GameDatabase.class,
                DB_NAME
        ).build();
    }

    @Provides
    @Singleton
    GameDao provideGameDao(GameDatabase gameDatabase) {
        return gameDatabase.gameDao();
    }
}

File: app/src/main/java/com/example/gamelister/model/db/RoomModule.kt
package com.example.gamelister.model.db

import android.arch.persistence.room.Room
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class RoomModule {
    companion object {
        private const val DB_NAME = "game-db"
    }

    @Provides
    @Singleton
    fun provideGameDatabase(context: Context): GameDataBase {
        return Room.databaseBuilder(
            context,
            GameDataBase::class.java,
            DB_NAME
        ).build()
    }

    @Provides
    @Singleton
    fun provideGameDao(gameDataBase: GameDataBase): GameDao = gameDataBase.gameDao()
}
